File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/stream/retrofit/FFMpegRequestBodyConverter.java
package org.dvbviewer.controller.data.stream.retrofit;

import org.dvbviewer.controller.entities.FFMpegPresetList;

import okhttp3.MediaType;
import okhttp3.RequestBody;
import retrofit2.Converter;

final class FFMpegRequestBodyConverter implements Converter<FFMpegPresetList, RequestBody> {

    private static final MediaType MEDIA_TYPE = MediaType.parse("text");

    @Override public RequestBody convert(FFMpegPresetList value) {
        return RequestBody.create(MEDIA_TYPE, value.toString());
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/stream/retrofit/FFMpegRequestBodyConverter.kt
package org.dvbviewer.controller.data.stream.retrofit

import okhttp3.MediaType
import okhttp3.RequestBody
import org.dvbviewer.controller.data.entities.FFMpegPresetList
import retrofit2.Converter

internal class FFMpegRequestBodyConverter : Converter<FFMpegPresetList, RequestBody> {

    override fun convert(value: FFMpegPresetList): RequestBody {
        return RequestBody.create(MEDIA_TYPE, value.toString())
    }

    companion object {

        private val MEDIA_TYPE = MediaType.parse("text")
    }

}